src/main/java/com/yelp/nrtsearch/server/luceneserver/field/properties/TermQueryable.java

```diff
@@ -25,18 +26,130 @@
  */
 public interface TermQueryable {
   /**
-   * Build a term query for this field type with the given configuration.
+   * Performs type validation. Do not @Override in subclasses.
+   *
+   * <p>Build a term query for this field type with the given configuration.
```

Comment:
Nit: can you switch the order of these lines. The first sentence becomes the summary line for generated java docs, so it should describe the functionality